### DISCUSSION TITLE
Detect Input Change

### DISCUSSION THREAD

**Discussion by joaopalopes24** [livewire/livewire#5332] [1. Help]
I have two number inputs, like this:

```
<input type="number" wire:model="value" />

<input wire:change="callFunction" x-on:change="call_function" type="number" wire:model="value" />
```

If I change the first input, the second is changed normally, but the change event is not read via alpine or via livewire. How do I get this to work normally and events to be read?

**Comment by Martleby** [comment]
One possibility would be to dispatch a browser event from your component - inside an updatedValue function - which you could then listen for with Alpine browser side. 

There are some other options, but it depends I think on what you want to do in the real world.

**Reply by joaopalopes24** [reply]
@Martleby I wanted to do something that didn't involve any functionality within the Livewire Component, preferably just using Alpine.

**Comment by joshhanley** [comment]
@joaopalopes24 `wire:change` and `x-on:change` will only fire if the input itself has changed (therefore firing the `change` event). This won't fire on a value change controlled by Livewire.

I would probably use Alpine to bind the second input and use `@entangle` for `value`, that way you can then setup an Alpine `$watch` to watch for changes and do whatever you need then.

Hope this helps!
